tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/IFindReplaceUIAccess.java

```diff
@@ -0,0 +1,59 @@
+package org.eclipse.ui.workbench.texteditor.tests;
+
+import java.util.Set;
+
+import org.eclipse.swt.widgets.Widget;
+
+import org.eclipse.jface.text.IFindReplaceTarget;
+
+import org.eclipse.ui.internal.findandreplace.IFindReplaceLogic;
+import org.eclipse.ui.internal.findandreplace.SearchOptions;
+
+
+/**
+ * Wraps UI-Access for different Find/Replace-UIs
+ */
+interface IFindReplaceUIAccess {
+
+	IFindReplaceTarget getTarget();
+
+	void close();
+
+	void ensureHasFocusOnGTK();
+
+	void unselect(SearchOptions option);
+
+	void select(SearchOptions option);
+
+	void simulateEnterInFindInputField(boolean shiftPressed);
+
+	void simulateKeyPressInFindInputField(int keyCode, boolean shiftPressed);
+
+	String getFindText();
+
+	String getReplaceText();
+
+	void setFindText(String text, boolean grabFocus);
+
+	void setReplaceText(String text);
+
+	Widget getButtonForSearchOption(SearchOptions option);
+
+	Set<SearchOptions> getEnabledOptions();
+
+	Set<SearchOptions> getSelectedOptions();
+
+	IFindReplaceLogic getFindReplaceLogic();
+
+	void pressReplaceAll();
+
+	/**
+	 * Some UIs require a "find" before allowing a Replace. This method handles that, ensuring that
+	 * if "findstring" is contained in the target's text, it will definitely be replaced.
+	 *
+	 * To keep parity between tests, this method should only be called if no text is currently
+	 * selected on the target.
+	 */
```

Comment:
There seems to be some important part in this specification, but the fact that different UIs may have different requirements for performing a replace is an implementation detail of the class realizing this interface. The specification should only state pre- and postconditions, i.e., what the caller of the method has to fulfill (no text must be selected on the target) and what the caller can expect (replacement of the specified find string in the target's text, if existing).

Reply:
changed to 
```java
	/**
	 * This method replaces all occurences of the find text in the arget with the replace text. No
	 * text must be selected on the target before calling.
	 */
```

https://github.com/eclipse-platform/eclipse.platform.ui/commit/49851d64fd5a67634b90f951a0aeddf3da74bfaa